feature/admin_page

Mashrur25 opened the pull request:
Added admin, group and group generation HTML. Added verification for only admins to create groups. Added randomised group generation. By Mashrur.

Only admins can access the admin page and see all the list of users and create groups for the 1st time. and after that if we have more than 4 or 5 members without groups then it will create groups otherwise not.

kbhv0880 committed:
Added admin, group and group generation HTML. Added verification for only admins to create groups. Added randomised group generation. By Mashrur.

ArseniyVasilko commented:
Haven't tested the code, or checked confilicts, but these things need to be looked into from general testing (highest to lowest importance order):

Implement jinja2 logic so that Admin Account and View ALL Groups only appears in Admins' navbars

Add a flash message notifying the admin that an x number of users could not be assigned a group due to a specific reason (e.g. not enough mentors: x more necessary, or uneven number of students (e.g. you can't split 7 students into 2 groups of 5))

When you press the group generation button with no Mentors present, the app displays "Groups Successfully Generated" message, even though no groups are created. Change to an error message in that case instead?

Change User Registeration Status text from True/False to Registered/Not Registered?
Grey out or otherwise change the Group Generation button once the groups have been generated?
Change View ALL groups tot View Groups? Seems a bit nicer to me.

Logeshsharma committed:
Merge remote-tracking branch 'origin/main' into feature/admin

# Conflicts:
#	app/data/data.sqlite
#	app/debug_utils.py
#	app/views.py

Logeshsharma committed:
Resolve conflict

ArseniyVasilko commented on app/templates/base.html:
```diff
@@ -36,6 +36,14 @@
                     <a class="nav-link" aria-current="page" href="{{ url_for('logout') }}">Logout {{
                         current_user.username }}</a>
                 </li>
+                    {% if current_user.registered %}
```
change from current_user.registered to current_user.role == "Admin", so that registered students would not be able to see this. (Technically they cannot anyway since the server -side will deny login to all non-admins in the main version, but still good practice)

Mashrur25 commented:
> Add a flash message notifying the admin that an x number of users could not be assigned a group due to a specific reason (e.g. not enough mentors: x more necessary, or uneven number of students (e.g. you can't split 7 students into 2 groups of 5))
> 
> When you press the group generation button with no Mentors present or no Students present, the app displays "Groups Successfully Generated" message, even though no groups are created. Change to an error message in that case instead?
> 
> Change User Registeration Status text from True/False to Registered/Not Registered? Grey out or otherwise change the Group Generation button once the groups have been generated? Change View ALL groups tot View Groups? Seems a bit nicer to me.

OK will try to incorporate these changes.

ArseniyVasilko commented on app/templates/admin_account.html:
```diff
@@ -0,0 +1,49 @@
+
+    <p>Hello {{ current_user.username }}</p>
+
+    <p>This page is only available to you as the currently logged in user</p>
```
Unnecessary line from the original account page?

ArseniyVasilko commented on app/templates/group.html:
```diff
@@ -0,0 +1,54 @@
+{% extends "base.html" %}
+{% import "bootstrap_wtf.html" as wtf %}
+
```
Perhaps rename group.html, /group and group() function to "groups" instead to avoid confusion? (group kind of implies on group and its details imo)

ArseniyVasilko commented on app/templates/group.html:
```diff
@@ -0,0 +1,54 @@
+    <p>This page is only available to you as the currently logged in user</p>
+
+    <p>The email address we have recorded for you is: {{ current_user.email }}</p>
+
```
Perhaps unnecessary given that it's not an account page and the literal same text and info can be found on the admin_account.html? consider deleting

ArseniyVasilko commented on app/templates/group.html:
```diff
@@ -0,0 +1,54 @@
+                        </tr>
+                    {% endfor %}
+                    </tbody>
+                </table>
```
table functions well, but the visual style is really difficult to comprehend with black lines on black, maybe change the styling/colours slightly?

ArseniyVasilko commented on app/templates/group_generation.html:
```diff
@@ -0,0 +1,49 @@
+            {% for group in all_groups %}
+                <tr>
+                    <td>{{ group.id }}</td>
+                       {% for user in group.users %}
```
Isn't this whole page essentially a copy of the group.html page? consider deleting this page and redirecting to group.html page when necessary (make changes in view.py accodringly)

Mashrur25 committed:
I added the check to see if 4 students and 1 mentor is present or not. fixed the fash message a little. By Mashrur.

Mashrur25 committed:
Modified the group generatio to display appropriate error messages. deleted the group_generation.html from the previous version. deleted the debug print statements. By Mashrur.

Logeshsharma committed:
Merge branch 'main' into feature/admin

# Conflicts:
#	app/data/data.sqlite
#	app/debug_utils.py
#	app/views.py

Logeshsharma committed:
Resolve conflict

Logeshsharma committed:
Update DB

ArseniyVasilko committed:
Minor changes to flash error message content and logic in "/group_generation"
Unsuccessful group creation attempts now redirect you the /admin_account page rather than /home
changed /group address to /groups everywhere and refactored accordingly.
